### DISCUSSION TITLE
Spring-boot runner

### DISCUSSION THREAD

**Discussion by javatechno** [apache/camel-karavan#1335] [Q&A] [answered]
### Ask a question

Could you exmplain, please, why exactly camel.jbang.runtime=spring-boot option is disabled in Karavan? Are there any specific design limitations for it? As I've checked, Camel-Jbang added quite a lot of support features for it in May in 4.6.0 version of Apache Camel.

Is the direction of adding spring-boot runner perspective for this project? If so, would you accept contributions?

### Variant

Web Application

### Container Management (if applicable)

Docker

### Operating System (if applicable)

Windows

### Version

4.7.0

**Comment by mgubaidullin** [comment, marked as answer]
Karavan running application with camel-jbang and camel-main in developer mode.
And camel-main exposes control, console, metrics and health api that is different than in sprong-boot and quarkus.
In addition to that bean configuration and properties are different in these three runtime.

To avoid confusing and having unpredictable difference in dev-mode and build application,
we mostly focus on camel-main runtime support.
